ports/samd/samd_isr.c

```diff
@@ -97,8 +100,18 @@ void SysTick_Handler(void) {
     if (soft_timer_next == next_tick) {
         pendsv_schedule_dispatch(PENDSV_DISPATCH_SOFT_TIMER, soft_timer_handler);
     }
+    #if defined(MCU_SAMD21)
+    rng_seed = (rng_seed * 17) ^ (REG_TC4_COUNT32_COUNT >> 1);
```

Comment:
You might want to put this at the very start of this function, to make it more deterministic when it is run (eg not dependent on the soft-timer check above).

Also, would be worth a comment, eg "use phase jitter between clocks ..."

--------

Further thought: having this code here run unconditionally is a small but non-trivial burden on the MCU.  It would be nice if it could be done only conditionally, or all the work done in `get_rng_seed`, but I can't see an easy way to do it because it needs the precision of the systick IRQ.

If the ISR table was in RAM then you could temporarily redirect `SysTick_Handler` to generate the seed... but that's going to waste a lot of RAM.